app/views/course/lectures/_lecture.html.slim

```diff
@@ -0,0 +1,43 @@
+- edit_button_path = [current_course, lecture]
+- delete_button_path = [current_course, lecture]
+
+= div_for(lecture, class: time_period_class(lecture) + unread_class(lecture))
+  h2.lecture-title
+    = format_inline_text(lecture.title)
+
+    - unless lecture.currently_active?
+      small title=time_period_message(lecture)
+        =< fa_icon 'calendar'.freeze
+
+    div.pull-right
+      div.btn-group
+        = edit_button(edit_button_path) if can?(:edit, lecture)
+        = delete_button(delete_button_path) if can?(:destroy, lecture)
+
+  i.timestamp
+    => format_datetime(lecture.start_at)
+    ' -
+    => format_datetime(lecture.end_at)
+    ' by
+    = link_to_user(lecture.creator)
+
+  div.content
+    = format_html(lecture.content)
+
+  div.access-link
+    - link_exists = lecture.instructor_classroom_link.present?
+    - is_instructor = can?(:edit, lecture)
+    - if lecture.currently_active?
```

Comment:
We have a self-directed learning mode which allows the student access future assignment, should that affect the starting time here ? 

some reference here: https://github.com/Coursemology/coursemology2/blob/master/app/models/course.rb#L121
and you can find the settings under course settings tab.